xapian-applications/omega/handler_libarchive.cc

```diff
@@ -55,68 +268,30 @@ extract(const string& filename,
 	    return false;
 	}
 
-	size_t total;
-	ssize_t size;
-	string s, s_content, s_styles;
-	struct archive_entry* entry;
-
-	// extracting data from content.xml, styles.xml and meta.xml
-	while (archive_read_next_header(archive_obj, &entry) == ARCHIVE_OK) {
-	    string pathname = archive_entry_pathname(entry);
-	    if (pathname == "content.xml") {
-		total = archive_entry_size(entry);
-		s_content.resize(total);
-		size = archive_read_data(archive_obj, &s_content[0], total);
-
-		if (size <= 0) {
-		    error = "Libarchive was not able to extract data from "
-			    "content.xml";
-		    return false;
-		}
-
-		s_content.resize(size);
-	    } else if (pathname == "styles.xml") {
-		total = archive_entry_size(entry);
-		s_styles.resize(total);
-		size = archive_read_data(archive_obj, &s_styles[0], total);
-
-		if (size <= 0) {
-		    error = "Libarchive was not able to extract data from "
-			    "styles.xml";
-		    return false;
-		}
-
-		s_styles.resize(size);
-	    } else if (pathname == "meta.xml") {
-		total = archive_entry_size(entry);
-		string s_meta;
-		s_meta.resize(total);
-		size = archive_read_data(archive_obj, &s_meta[0], total);
-
-		if (size > 0) {
-		    // indexing file even if this fails
-		    s_meta.resize(size);
-		    MetaXmlParser metaxmlparser;
-		    metaxmlparser.parse(s_meta);
-		    title = metaxmlparser.title;
-		    keywords = metaxmlparser.keywords;
-		    author = metaxmlparser.author;
-		}
-	    }
+	bool succeed;
+	if (startswith(mimetype, "application/vnd.sun.xml.") ||
+	    startswith(mimetype, "application/vnd.oasis.opendocument.")) {
+	    succeed = extract_odf(archive_obj, dump, title, keywords, author,
+				  error);
+	    if (!succeed)
+		return false;
+	} else if (startswith(mimetype, "application/"
+					"vnd.openxmlformats-officedocument.")) {
```

Comment:
It would be better to wrap the line by aligning the function parameters vertically rather than splitting the string - that's more typical of how this sort of case is wrapped in existing code in the codebase, it seems easier to read that way, and it means things like `grep application/vnd` match here.

Reply:
I have changed this. But in order to maintain 80 chars per line, I had to move the opening bracket of the else if statement to the next line (against the usual practice). Will that be ok?

Reply:
Yes, I think that's OK.

When it comes down to it, the point of having such formatting rules as part of coding standards is mainly to aid readability and make it easier to work with the source code, and to help spot problems (if things are consistent then inconsistencies stand out, and bugs are often due to inconsistencies).